### PR TITLE
piston-cli: 1.3.0 -> 1.4.1

### PR DESCRIPTION
**Pull request by ethancedwards8**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by fufexan — APPROVED** [review]

File: pkgs/tools/misc/piston-cli/default.nix [hunk 1]
```diff
@@ -1,15 +1,15 @@
-{ lib, python3Packages }:
+{ lib, python39Packages }:
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/piston-cli/default.nix:1]: Please overwrite the passed in python3Packages.

Comment (ethancedwards8) [reply pkgs/tools/misc/piston-cli/default.nix:1]: I don't understand what you mean?

Comment (fufexan) [reply pkgs/tools/misc/piston-cli/default.nix:1]: In `pkgs/top-level/all-packages.nix`:
```nix
  piston-cli = callPackage ../tools/misc/piston-cli { python3Packages = python39Packages; };
```

**Review by fufexan — APPROVED** [review]

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 124531 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary>  <ul>    <li>piston-cli</li>  </ul></details>

**Pull request merged**